### PR TITLE
Upgrade prettier and eslint compatibility libraries to latest

### PR DESCRIPTION
**Pull request by ozyx**
Closes 

### Describe your changes:

In order to upgrade to prettier 3.x.x, we also need to upgrade the related eslint compatibility libraries (eslint, eslint-plugin-prettier, eslint-config-prettier). This PR does it all in one go, and runs lint:fix which only has one small change.

Also adds `.vscode/extensions.json` so that vscode can recommend extensions (such as prettier-eslint) for working in our codebase

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [ ] Tests included and/or updated with changes?
* [x] Has this been smoke tested?
* [x] Have you associated this PR with a `type:` label? Note: this is not necessarily the same as the original issue.
* [x] Have you associated a milestone with this PR? Note: leave blank if unsure.
* [ ] Is this a breaking change to be called out in the release notes?
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [ ] Changes appear to address issue?
* [ ] Reviewer has tested changes by following the provided instructions?
* [ ] Changes appear not to be breaking changes?
* [ ] Appropriate automated tests included?
* [ ] Code style and in-line documentation are appropriate?

### CODE REVIEW

**Comment by ozyx** [comment]
...and we have a new release that works! Huzzah!

File: .vscode/extensions.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+  // See https://go.microsoft.com/fwlink/?LinkId=827846 to learn about workspace recommendations.
```
Comment (unlikelyzero) [inline .vscode/extensions.json:2]: broken json

Comment (ozyx) [reply .vscode/extensions.json:2]: huh. i think it still works but i'll remove it anyway

File: src/plugins/DeviceClassifier/src/DeviceMatchersSpec.js [hunk 2]
```diff
@@ -46,14 +46,14 @@ describe('DeviceMatchers', function () {
...
-  });
+  ['mobile', 'phone', 'tablet', 'landscape', 'portrait', 'landscape', 'touch'].forEach(
+    function (deviceType) {
+      it('detects when a device is a ' + deviceType + ' device', function () {
+        mockAgent[method(deviceType)].and.returnValue(true);
+        expect(DeviceMatchers[deviceType](mockAgent)).toBe(true);
+        mockAgent[method(deviceType)].and.returnValue(false);
+        expect(DeviceMatchers[deviceType](mockAgent)).toBe(false);
+      });
+    }
```
Comment (unlikelyzero) [inline src/plugins/DeviceClassifier/src/DeviceMatchersSpec.js:57]: what is this

Comment (ozyx) [reply src/plugins/DeviceClassifier/src/DeviceMatchersSpec.js:57]: old code. and a small change in prettier's opinionated formatting

**Review by unlikelyzero — APPROVED** [review]

File: .vscode/extensions.json (hunk 1 above)
Comment (JounQin) [reply .vscode/extensions.json]: It's jsonc actually. Just like `tsconfig.json`.

**Pull request merged**